jsonrpc/src/test/testCancellationStrategy.ts

```diff
@@ -0,0 +1,136 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as fs from 'fs';
+import * as path from 'path';
+import { CancellationReceiverStrategy, CancellationId, CancellationSenderStrategy, MessageConnection, Emitter, Event, CancellationToken, AbstractCancellationTokenSource } from '../main';
```

Comment:
duplicated since I couldn't figure out how to share this file in tests.

Reply:
Should that test be in the new cancellation package ?

Reply:
it is no longer duplicated and I think we need at least 1 test that tests new CancellationStrategy in jsonrpc since it is defined in jsonrpc.

it doesn't need to be file based cancellation though. just custom implementation of CancellationStrategy.